src/schedule_synthesis.jl

```diff
@@ -0,0 +1,273 @@
+using RealTimeScheduling
+using DataStructures
+
+"""
+    schedule_xghtc(constraints; slotsize=1, H=100)
+
+Generate a schedule for a set of weakly hard constraints. The schedule assumes that all tasks are synchronous and have equal periods. At most `slotsize` tasks may be scheduled in a single period. The schedule has total length `H` periods.
+    
+Shengjie Xu, Bineet Ghosh, Clara Hobbs, P.S. Thiagarajan, and Samarjit Chakraborty, "Safety-Aware Flexible Schedule Synthesis for Cyber-Physical Systems using Weakly-Hard Constraints." ASP-DAC 2023.
```

Comment:
Please keep docstrings to a max of 92 characters per line. https://docs.julialang.org/en/v1/manual/documentation/

Also, it would be great to add the DOI link to the paper for quick access.